### DISCUSSION TITLE
AwoX 33952

### DISCUSSION THREAD

**Discussion by titou4307** [Koenkk/zigbee2mqtt#16233] [Devices]
Hello

I would like to pair my AwoX 33952 with zigbee2mqtt.....but impossible

Does anyone could give me the process for pair this remote controller ?

**Comment by titou4307** [comment]
Hi
I find a youtube video here : 

https://www.youtube.com/watch?v=KuVKFEIm4qg

**Comment by kzajac83** [comment]
Have the same problem, remote controler register not as AwoX 33952  but as "ERCU_Zm" 

Paring step finish success and the procedure is on pilot need 1st need update FW usisng AwoX orgiginal app via smarphone and BT and after flashing keep power button and colored circle with arrow button on bottom of pilot and waiting for 3. or 4. flash red LED  - this erase connection memory and allow next step. Is need keep power btn ~ 3s for estabils zigbee paring.

However total task fail due to unsported device discovered as "ERCU_Zm".  Due to not found solution I read this page:

https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

next found link to dev branch and found this:

https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/awox.js

To implement is need in folder Zigbee2Mqtt where is configuration.yaml file for Z2M insert above file and update configuration file for new  convertrs and restart Z2M.

Not sure above resolve the issue or not becase made this 1st time and chage many things in the same time so not sure help or not. 

If not help then in section "module.exports" inside square backets is need add:

` {
    zigbeeModel: ['ERCU_Zm'],
    model: '33952',
    vendor: 'AwoX',
    description: 'Remote RGB+CCT dim controller',
    fromZigbee: [fz.command_on, fzLocal.colors, fzLocal.refresh, fzLocal.refreshColored, fz.command_off,
            fz.command_step, fz.command_move, fz.command_stop, fz.command_recall, fz.command_step_color_temperature],
        toZigbee: [],
        exposes: [e.action(['on', 'off', 'red', 'refresh', 'refresh_colored', 'blue', 'yellow',
            'green', 'brightness_step_up', 'brightness_step_down', 'brightness_move_up', 'brightness_move_down', 'brightness_stop',
            'recall_1', 'color_temperature_step_up', 'color_temperature_step_down'])],
    },`

and change header of the file "const" becase due to path issue Z2M will not start after restart. 

General my file is in:

https://pastebin.com/qxfEAwZG

For me working but owner of AwoX.js file should implement better my code.

**Reply by Koenkk** [reply]
Added, changes will be included in the next release.

**Comment by kzajac83** [comment]
In HA I switch Zigbee2MQTT add-on to dev/Edge instance and from configuration erase my .js file and try add  next remote controller (I have 3 the same devices) and results below:

![image](https://user-images.githubusercontent.com/50861081/231998258-b414ab40-031f-4298-b3dc-753e712d67b7.png)

Probalby doing somthing wrong but download awox.js into Zigbee2MQTT folder and in settings of add-on for external converters add awox.js and restart Edge instace.  However back my old issue 

``
Zigbee2MQTT Edge
[11:10:05] INFO: Preparing to start...
[11:10:06] INFO: Socat not enabled
[11:10:07] INFO: Starting Zigbee2MQTT...
node:internal/modules/cjs/loader:1024
  throw err;
  ^
Error: Cannot find module '../lib/extend'
Require stack:
- /app/dist/util/utils.js
- /app/dist/util/settings.js
- /app/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1021:15)
    at Function.Module._load (node:internal/modules/cjs/loader:866:27)
    at Module.require (node:internal/modules/cjs/loader:1093:19)
    at require (node:internal/modules/cjs/helpers:108:18)
    at /app/data/extension/externally-loaded.js:1:16
    at Script.runInContext (node:vm:141:12)
    at Script.runInNewContext (node:vm:146:17)
    at Object.runInNewContext (node:vm:306:38)
    at loadModuleFromText (/app/lib/util/utils.ts:150:8)
    at loadModuleFromFile (/app/lib/util/utils.ts:157:12)
``


So my question is what I doing wrong?

PS. Regarding flashing remote controller small update - after flash  is need remove and insert batteries.

**Reply by Koenkk** [reply]
Try with https://gist.github.com/Koenkk/f3e9d9a77681dd628dac092c03f8d5b5

**Comment by kzajac83** [comment]
After add above file into config

![image](https://user-images.githubusercontent.com/50861081/232167798-23a0d52a-135c-4d11-b4f3-919e800f196f.png)

and put file into /config/zigbee2mqtt/ext.js is working fine after restart Z2M add-on
